src/components/map/Map.tsx

```diff
@@ -157,131 +144,264 @@ export const Map = ({
+          [-Infinity, -Infinity]
+        );
+        const deltaLng = (maxLng - minLng) * 0.1; // Add 10% padding to the bounds
+        const deltaLat = (maxLat - minLat) * 0.1; // Add 10% padding to the bounds
+        if (cameraRef.current) {
+          cameraRef.current.fitBounds(
+            [minLng - deltaLng, minLat - deltaLat],
+            [maxLng + deltaLng, maxLat + deltaLat],
+            0 // animationDuration in ms
+          );
+          if (clusteringEnabled) {
+            setTimeout(() => {
+              setIsLoading(false);
+            }, 1000); // Timeout in Millisekunden anpassen je nach Bedarf
+            setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+          } else {
+            setTimeout(() => {
+              setIsLoading(false);
+            }, 4000); // Timeout in Millisekunden anpassen je nach Bedarf
+            setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+          }
+        }
+      }
+    }
+  }, [locations, clusterDistance, zoomLevel, isInitialFit, clusteringEnabled]);
+
+  useEffect(() => {
+    if (updatedRegion && cameraRef.current) {
+      cameraRef.current.setCamera({
+        centerCoordinate: [updatedRegion.longitude, updatedRegion.latitude],
+        zoomLevel: 14,
+        animationDuration: 1000
+      });
+    }
+  }, [updatedRegion]);
+
+  const handleMapPress = (event) => {
+    if (onMapPress) {
+      const { geometry } = event;
+      const nativeEvent = {
+        coordinate: {
+          latitude: geometry.coordinates[1],
+          longitude: geometry.coordinates[0]
+        }
+      };
+      onMapPress({ nativeEvent });
+    }
+  };
+
+  const handleClusterPress = useCallback(
+    (cluster) => {
+      if (isAnimating) return; // Skip if animation is already in progress
+
+      setIsAnimating(true);
+      const [longitude, latitude] = cluster.geometry.coordinates;
+      const zoom = Math.min(zoomLevel + 2, 20); // Increase the zoom level by 2 or set to max zoom level (20)
+      if (cameraRef.current) {
+        cameraRef.current.setCamera({
+          centerCoordinate: [longitude, latitude],
+          zoomLevel: zoom,
+          animationDuration: 1000
+        });
+      }
+
+      // Wait for the animation to finish
+      setTimeout(() => {
+        setIsAnimating(false);
+      }, 1000);
+    },
+    [isAnimating, zoomLevel]
+  );
+
   return (
     <View style={[styles.container, style]}>
-      <MapView
-        clusterColor={colors.primary}
-        clusterFontFamily="regular"
-        clusteringEnabled={clusteringEnabled}
-        radius={clusterDistance}
-        renderCluster={renderCluster}
-        initialRegion={initialRegion}
-        region={updatedRegion}
-        mapType={device.platform === 'android' ? MAP_TYPES.NONE : MAP_TYPES.STANDARD}
-        onPress={onMapPress}
+      <LoadingSpinnerMap loading={isLoading} />
+      <MapLibreGL.MapView
+        style={mapStyle || styles.map}
```

Comment:
could there be cases where both styles should be used? like passing some additional styles per `mapStyle` to `styles.map` in some cases? therefore we could change to array with `mapStyle` last.

Reply:
thanks, will be adjusted in the upcoming commit.